projects/angular/src/data/datagrid/datagrid-column.ts

```diff
@@ -357,6 +341,19 @@ export class ClrDatagridColumn<T = any>
       return;
     }
 
+    if (reverse === undefined) {
+      let nextSort = ClrDatagridSortOrder.UNSORTED;
+
+      if (this._sortOrder === ClrDatagridSortOrder.UNSORTED) {
+        nextSort = ClrDatagridSortOrder.ASC;
+      } else if (this._sortOrder === ClrDatagridSortOrder.ASC) {
+        nextSort = ClrDatagridSortOrder.DESC;
+      }
+
+      this.doSort(nextSort);
+      return;
```

Comment:
Column button is using sort() only and because of the return here it would not emit events that we expose trough API  and also the other functions would not pass trough